### PR TITLE
Implement affine subspaces

### PR DESCRIPTION
**Pull request by JoeyBF**
This will make it easier to handle Massey products in the future. Extends #148

### CODE REVIEW

File: ext/crates/fp/src/matrix/subspace.rs [hunk 1]
```diff
@@ -209,18 +214,34 @@ impl Subspace {
...
-        for row in self.matrix.iter().take(dim) {
-            if f.alternate() {
-                writeln!(f, "{row:#}")?;
-            } else {
-                writeln!(f, "{row}")?;
-            }
-        }
+        let mut rows = self.matrix.iter().take(dim);
+
+        let output = if f.alternate() {
```
Comment (hoodmane) [inline ext/crates/fp/src/matrix/subspace.rs:234]: What does `f.alternate()` mean?

**Review by hoodmane — APPROVED** [review]
Thanks!

File: ext/crates/fp/src/matrix/subspace.rs (hunk 1 above)
Comment (JoeyBF) [reply ext/crates/fp/src/matrix/subspace.rs:234]: It formats the subspace in one line, to anticipate printing Massey products to `stdout`. I'll see if I can add a docstring.

Comment (hoodmane) [reply ext/crates/fp/src/matrix/subspace.rs:234]: Is there some choice of the name? What about calling it `one_line` or something a bit more explicit?

Comment (JoeyBF) [reply ext/crates/fp/src/matrix/subspace.rs:234]: It's a method on `std::fmt::Formatter` so we're stuck with that. In general for anything that implements Display you can print it normally using "{foo}" or use an alternate format with "{foo:#}".

Comment (JoeyBF) [reply ext/crates/fp/src/matrix/subspace.rs:253]: I added a doctest. Since it's part of the `Subspace` API it's also included in #148

**Pull request merged**